### PR TITLE
feat(api): add DecorationsRequest filter for build config

### PR DESCRIPTION
**Pull request by schroederc**

### CODE REVIEW

**Comment by schroederc** [comment]
R: @hildrum

File: kythe/go/serving/xrefs/xrefs.go [hunk 1]
```diff
@@ -292,6 +293,11 @@ func (t *Table) Decorations(ctx context.Context, req *xpb.DecorationsRequest) (*
 		bindings := stringset.New()
 
 		for _, d := range decor.Decoration {
+			// Filter decorations by requested build configs.
+			if len(buildConfigs) != 0 && !buildConfigs.Contains(d.Anchor.BuildConfiguration) {
+				continue
+			}
+
```
Comment (jaysachs) [inline kythe/go/serving/xrefs/xrefs.go]: This filter decorations. Isn't it possible that asking for xrefs for a ticket might also need filtering? (i.e. if the xref is in some other build config)?

Comment (schroederc) [reply kythe/go/serving/xrefs/xrefs.go]: Filtering for xrefs will be done in a separate change.

**Comment by hildrum** [comment]
LGTM, though you don't appear to have a test where the request has an empty build config and there decorations returned for all build configs.

File: kythe/proto/xref.proto [hunk 2]
```diff
@@ -179,6 +184,9 @@ message DecorationsReply {
     // The reference's span within the source file.
     common.Span span = 5;
 
+    // The build config of the file containing this reference.
+    string build_config = 6;
```
Comment (creachadair) [inline kythe/proto/xref.proto]: Couldn't more than one configuration apply?

Comment (schroederc) [reply kythe/proto/xref.proto]: Each anchor has a single build configuration.

**Comment by schroederc** [comment]
> LGTM, though you don't appear to have a test where the request has an empty build config and there decorations returned for all build configs.

The existing tests cover the empty filter case.

**Pull request merged**